scripts/qemu.py

```diff
@@ -135,13 +135,16 @@ def launch(self):
             self._popen = subprocess.Popen(args, stdin=devnull, stdout=qemulog,
                                            stderr=subprocess.STDOUT, shell=False)
             self._post_launch()
-        except:
+        except Exception as details:
+            self._load_io_log()
             if self.is_running():
                 self._popen.kill()
                 self._popen.wait()
-            self._load_io_log()
+            else:
+                details.message += ("\nCommand:\n%s\nOutput:\n%s" %
+                                    (' '.join(args), self._iolog))
             self._post_shutdown()
-            raise
+            raise details.__class__(details.message)
```

Comment:
This cripples some exceptions (eg. IOError which does not store the message in `message`) and might break other custom exceptions which don't accept message as first argument. I'd recommend keeping the `raise` here and in case it was not running I'd raise a new exception explaining the original one and adding the note about command and output.